Compile bug: Vulkan backend, glslc shader compilation errors (Ubuntu 24.04.3)

### Git commit

git rev-parse HEAD is 
dcca0d3ab840ebe9b2ccd4719033d408eeb758d7



### Operating systems

Linux

### GGML backends

Vulkan

### Problem description & steps to reproduce


Bulding the Vulkan backend gives shader compilation errors for rope_neox_f16, rope_norm_f16, rope_norm_f16_rt,  rope_neox_f16_rte: 

```
/bin/glslc -fshader-stage=compute --target-env=vulkan1.2 -O /opt/llama.cpp/llama.cpp/ggml/src/ggml-vulkan/vulkan-shaders/rop
e_neox.comp -o /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders.spv/rope_neox_f16.spv -DA_TYPE=float16_t -DD_
TYPE=float16_t

shaderc: internal error: compilation succeeded but failed to optimize: Expected input to have different bit width from Resul
t Type: FConvert
  %221 = OpFConvert %half %220
```

etc..

and subsequent linking errors: 
```
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_data'
collect2: error: ld returned 1 exit status
```

Steps to reproduce: 
* check out HEAD (i.e dcca0d3ab840ebe9b2ccd4719033d408eeb758d7)
* configure and build Vulkan backend in an empty directory on Ubuntu 24.04.3.
 
I am using the distribution supplied version of glslc: 
```
$ glslc --version
shaderc 2023.8-1build1
spirv-tools 2023.6~rc1-2
glslang 14.0.0-2

Target: SPIR-V 1.0

$ dpkg -s glslc
Package: glslc
Status: install ok installed
Priority: optional
Section: libdevel
Installed-Size: 196
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: foreign
Source: shaderc
Version: 2023.8-1build1
Depends: libc6 (>= 2.38), libgcc-s1 (>= 3.3.1), libshaderc1 (>= 2022.2-1), libstdc++6 (>= 13.1)
Description: Command line compiler for GLSL/HLSL to SPIR-V
 glslc wraps around core functionality in glslang and SPIRV-Tools. glslc and
 its library aims to provide:
    - a command line compiler with GCC- and Clang-like usage, for better
    integration with build systems;
    - an API where functionality can be added without breaking existing clients;
    - an API supporting standard concurrency patterns across multiple operating
    systems;
    - increased functionality such as file #include support.
 .
 This package contains the glslc binary.
Built-Using: glslang (= 14.0.0-2), spirv-tools (= 2023.6~rc1-2)
Homepage: https://github.com/google/shaderc/
Original-Maintainer: Philippe SWARTVAGHER <phil.swart@gmx.fr>
```


### First Bad Commit

* last working commit: 3464bdac37027c5e9661621fc75ffcef3c19c6ef
* braking commit: b9ce94017729465895402cbcfffb51fa926c15e3


### Compile command

```shell
(cmake ../llama.cpp/ -DGGML_VULKAN=ON -DCMAKE_BUILD_TYPE=Release && make -j) >log.txt 2>&1
```

### Relevant log output

```shell
CMAKE_BUILD_TYPE=Release
-- Warning: ccache not found - consider installing it for faster compilation or disable this warning with GGML_CCACHE=OFF
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- GGML_SYSTEM_ARCH: x86
-- Including CPU backend
-- x86 detected
-- Adding CPU backend variant ggml-cpu: -march=native 
-- Vulkan found
-- GL_KHR_cooperative_matrix supported by glslc
-- GL_NV_cooperative_matrix2 not supported by glslc
-- GL_EXT_integer_dot_product not supported by glslc
-- GL_EXT_bfloat16 not supported by glslc
-- Including Vulkan backend
-- ggml version: 0.9.4
-- ggml commit:  dcca0d3ab
-- Configuring done (1.0s)
-- Generating done (0.3s)
-- Build files have been written to: /opt/llama.cpp/build-vulkan
[  0%] Performing configure step for 'vulkan-shaders-gen'
[  1%] Built target llama-gemma3-cli
[  1%] Built target sha256
[  2%] Building C object examples/gguf-hash/CMakeFiles/xxhash.dir/deps/xxhash/xxhash.c.o
[  2%] Built target sha1
[  2%] Built target build_info
[  2%] Built target llama-llava-cli
[  3%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml.c.o
[  3%] Built target llama-qwen2vl-cli
[  4%] Built target llama-minicpmv-cli
[  4%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/gguf.cpp.o
[  4%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-alloc.c.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-backend.cpp.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-opt.cpp.o
[  5%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-quants.c.o
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Enabling coopmat glslc support
-- Configuring done (0.6s)
-- Generating done (0.0s)
-- Build files have been written to: /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders-gen-prefix/src/vulkan-shaders-gen-build
[  5%] Performing build step for 'vulkan-shaders-gen'
[ 50%] Building CXX object CMakeFiles/vulkan-shaders-gen.dir/vulkan-shaders-gen.cpp.o
[  5%] Built target xxhash
[  6%] Linking CXX shared library ../../bin/libggml-base.so
[  6%] Built target ggml-base
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/hbm.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.cpp.o
[  6%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.c.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/binary-ops.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/mmq.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/amx.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/traits.cpp.o
[  7%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/arch/x86/quants.c.o
[  7%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/quants.c.o
[  7%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ops.cpp.o
[  7%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/repack.cpp.o
[  8%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/arch/x86/repack.cpp.o
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/unary-ops.cpp.o
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/vec.cpp.o
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/llamafile/sgemm.cpp.o
[100%] Linking CXX executable vulkan-shaders-gen
[100%] Built target vulkan-shaders-gen
[  9%] Performing install step for 'vulkan-shaders-gen'
-- Installing: /opt/llama.cpp/build-vulkan/Release/./vulkan-shaders-gen
[  9%] Completed 'vulkan-shaders-gen'
[ 10%] Built target vulkan-shaders-gen
[ 10%] Generate vulkan shaders for wkv7.comp
[ 10%] Generate vulkan shaders for acc.comp
[ 10%] Generate vulkan shaders for clamp.comp
[ 10%] Generate vulkan shaders for conv2d_mm.comp
[ 10%] Generate vulkan shaders for argsort.comp
[ 11%] Generate vulkan shaders for copy_from_quant.comp
[ 11%] Generate vulkan shaders for copy.comp
[ 11%] Generate vulkan shaders for add.comp
[ 11%] Generate vulkan shaders for conv_transpose_1d.comp
[ 12%] Generate vulkan shaders for add_id.comp
[ 12%] Generate vulkan shaders for concat.comp
[ 12%] Generate vulkan shaders for argmax.comp
[ 13%] Generate vulkan shaders for contig_copy.comp
[ 13%] Generate vulkan shaders for cos.comp
[ 13%] Generate vulkan shaders for count_equal.comp
[ 14%] Generate vulkan shaders for dequant_iq1_m.comp
[ 14%] Generate vulkan shaders for dequant_iq1_s.comp
[ 14%] Generate vulkan shaders for dequant_iq2_xs.comp
[ 15%] Generate vulkan shaders for dequant_iq3_s.comp
[ 15%] Generate vulkan shaders for copy_to_quant.comp
[ 15%] Generate vulkan shaders for dequant_iq4_xs.comp
[ 15%] Generate vulkan shaders for dequant_f32.comp
[ 15%] Generate vulkan shaders for dequant_iq4_nl.comp
[ 15%] Generate vulkan shaders for dequant_iq2_s.comp
[ 15%] Generate vulkan shaders for dequant_iq2_xxs.comp
[ 15%] Generate vulkan shaders for dequant_iq3_xxs.comp
[ 16%] Generate vulkan shaders for dequant_q2_k.comp
[ 16%] Generate vulkan shaders for dequant_q4_1.comp
[ 16%] Generate vulkan shaders for dequant_q4_0.comp
[ 16%] Generate vulkan shaders for dequant_mxfp4.comp
[ 16%] Generate vulkan shaders for conv2d_dw.comp
[ 16%] Generate vulkan shaders for dequant_q4_k.comp
[ 17%] Generate vulkan shaders for dequant_q5_0.comp
[ 17%] Generate vulkan shaders for dequant_q5_k.comp
[ 17%] Generate vulkan shaders for dequant_q5_1.comp
[ 17%] Generate vulkan shaders for dequant_q3_k.comp
[ 17%] Generate vulkan shaders for dequant_q8_0.comp
[ 17%] Generate vulkan shaders for div.comp
[ 17%] Generate vulkan shaders for geglu.comp
[ 18%] Generate vulkan shaders for diag_mask_inf.comp
[ 18%] Generate vulkan shaders header
[ 18%] Generate vulkan shaders for dequant_q6_k.comp
[ 18%] Generate vulkan shaders for exp.comp
[ 18%] Generate vulkan shaders for geglu_quick.comp
[ 18%] Generate vulkan shaders for flash_attn_cm1.comp
[ 19%] Generate vulkan shaders for l2_norm.comp
[ 19%] Generate vulkan shaders for gelu_quick.comp
[ 19%] Generate vulkan shaders for leaky_relu.comp
[ 20%] Generate vulkan shaders for flash_attn_cm2.comp
[ 20%] Generate vulkan shaders for geglu_erf.comp
[ 21%] Generate vulkan shaders for gelu.comp
[ 21%] Generate vulkan shaders for flash_attn.comp
[ 21%] Generate vulkan shaders for hardswish.comp
[ 21%] Generate vulkan shaders for get_rows.comp
[ 21%] Generate vulkan shaders for flash_attn_split_k_reduce.comp
[ 21%] Generate vulkan shaders for im2col.comp
[ 21%] Generate vulkan shaders for mul.comp
[ 21%] Generate vulkan shaders for gelu_erf.comp
[ 21%] Generate vulkan shaders for get_rows_quant.comp
[ 21%] Generate vulkan shaders for mul_mat_split_k_reduce.comp
[ 22%] Generate vulkan shaders for group_norm.comp
[ 22%] Generate vulkan shaders for mul_mat_vec.comp
[ 22%] Generate vulkan shaders for hardsigmoid.comp
[ 22%] Generate vulkan shaders for mul_mat_vec_iq1_s.comp
[ 21%] Generate vulkan shaders for im2col_3d.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_iq1_m.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_iq2_s.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_iq2_xs.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_iq3_xxs.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_iq2_xxs.comp
[ 23%] Generate vulkan shaders for mul_mat_vec_p021.comp
[ 24%] Generate vulkan shaders for mul_mat_vec_q3_k.comp
[ 24%] Generate vulkan shaders for mul_mat_vec_nc.comp
[ 25%] Generate vulkan shaders for mul_mat_vec_iq3_s.comp
[ 25%] Generate vulkan shaders for mul_mat_vec_q2_k.comp
[ 25%] Generate vulkan shaders for mul_mat_vec_q5_k.comp
[ 25%] Generate vulkan shaders for mul_mat_vecq.comp
[ 25%] Generate vulkan shaders for mul_mat_vec_q6_k.comp
[ 25%] Generate vulkan shaders for mul_mat_vec_q4_k.comp
[ 25%] Generate vulkan shaders for mul_mm_cm2.comp
[ 25%] Generate vulkan shaders for norm.comp
[ 26%] Generate vulkan shaders for reglu.comp
[ 27%] Generate vulkan shaders for mul_mm.comp
[ 27%] Generate vulkan shaders for pad.comp
[ 27%] Generate vulkan shaders for mul_mmq.comp
[ 27%] Generate vulkan shaders for repeat.comp
[ 27%] Generate vulkan shaders for rms_norm.comp
[ 27%] Generate vulkan shaders for pool2d.comp
[ 27%] Generate vulkan shaders for multi_add.comp
[ 28%] Generate vulkan shaders for rms_norm_back.comp
[ 28%] Generate vulkan shaders for quantize_q8_1.comp
[ 28%] Generate vulkan shaders for roll.comp
[ 28%] Generate vulkan shaders for repeat_back.comp
[ 28%] Generate vulkan shaders for relu.comp
[ 27%] Generate vulkan shaders for rope_multi.comp
[ 28%] Generate vulkan shaders for opt_step_sgd.comp
[ 29%] Generate vulkan shaders for opt_step_adamw.comp
[ 29%] Generate vulkan shaders for rope_neox.comp
[ 29%] Generate vulkan shaders for scale.comp
[ 30%] Generate vulkan shaders for rope_norm.comp
[ 30%] Generate vulkan shaders for silu.comp
[ 30%] Generate vulkan shaders for rope_vision.comp
[ 29%] Generate vulkan shaders for rms_norm_partials.comp
[ 30%] Generate vulkan shaders for sigmoid.comp
[ 30%] Generate vulkan shaders for sqrt.comp
[ 31%] Generate vulkan shaders for silu_back.comp
[ 31%] Generate vulkan shaders for sin.comp
[ 32%] Generate vulkan shaders for square.comp
[ 32%] Generate vulkan shaders for soft_max.comp
[ 32%] Generate vulkan shaders for soft_max_back.comp
[ 32%] Generate vulkan shaders for ssm_conv.comp
[ 32%] Generate vulkan shaders for timestep_embedding.comp
[ 33%] Generate vulkan shaders for sub.comp
[ 34%] Generate vulkan shaders for upscale.comp
[ 33%] Generate vulkan shaders for sum_rows.comp
[ 33%] Generate vulkan shaders for swiglu.comp
[ 34%] Generate vulkan shaders for tanh.comp
[ 34%] Generate vulkan shaders for ssm_scan.comp
[ 34%] Generate vulkan shaders for swiglu_oai.comp
[ 34%] Generate vulkan shaders for topk_moe.comp
[ 34%] Generate vulkan shaders for wkv6.comp
cannot compile rope_neox_f16

/bin/glslc -fshader-stage=compute --target-env=vulkan1.2 -O /opt/llama.cpp/llama.cpp/ggml/src/ggml-vulkan/vulkan-shaders/rope_neox.comp -o /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders.spv/rope_neox_f16.spv -DA_TYPE=float16_t -DD_TYPE=float16_t 

shaderc: internal error: compilation succeeded but failed to optimize: Expected input to have different bit width from Result Type: FConvert
  %221 = OpFConvert %half %220



cannot compile rope_norm_f16

/bin/glslc -fshader-stage=compute --target-env=vulkan1.2 -O /opt/llama.cpp/llama.cpp/ggml/src/ggml-vulkan/vulkan-shaders/rope_norm.comp -o /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders.spv/rope_norm_f16.spv -DA_TYPE=float16_t -DD_TYPE=float16_t 

shaderc: internal error: compilation succeeded but failed to optimize: Expected input to have different bit width from Result Type: FConvert
  %212 = OpFConvert %half %211



cannot compile rope_norm_f16_rte

/bin/glslc -fshader-stage=compute --target-env=vulkan1.2 -O /opt/llama.cpp/llama.cpp/ggml/src/ggml-vulkan/vulkan-shaders/rope_norm.comp -o /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders.spv/rope_norm_f16_rte.spv -DA_TYPE=float16_t -DD_TYPE=float16_t -DRTE16=1 

shaderc: internal error: compilation succeeded but failed to optimize: Expected input to have different bit width from Result Type: FConvert
  %212 = OpFConvert %half %211



cannot compile rope_neox_f16_rte

/bin/glslc -fshader-stage=compute --target-env=vulkan1.2 -O /opt/llama.cpp/llama.cpp/ggml/src/ggml-vulkan/vulkan-shaders/rope_neox.comp -o /opt/llama.cpp/build-vulkan/ggml/src/ggml-vulkan/vulkan-shaders.spv/rope_neox_f16_rte.spv -DA_TYPE=float16_t -DD_TYPE=float16_t -DRTE16=1 

shaderc: internal error: compilation succeeded but failed to optimize: Expected input to have different bit width from Result Type: FConvert
  %221 = OpFConvert %half %220



[ 34%] Linking CXX shared library ../../bin/libggml-cpu.so
[ 34%] Built target ggml-cpu
[ 34%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/add_id.comp.cpp.o
[ 35%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/add.comp.cpp.o
[ 35%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/acc.comp.cpp.o
[ 35%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o
[ 36%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/concat.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/copy.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/argmax.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/conv2d_mm.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/cos.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/contig_copy.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/conv2d_dw.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/count_equal.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/copy_to_quant.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/argsort.comp.cpp.o
[ 37%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/copy_from_quant.comp.cpp.o
[ 38%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_f32.comp.cpp.o
[ 38%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/clamp.comp.cpp.o
[ 38%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/conv_transpose_1d.comp.cpp.o
[ 38%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq1_s.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq2_xxs.comp.cpp.o
[ 38%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq1_m.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq3_xxs.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq4_xs.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq2_xs.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq4_nl.comp.cpp.o
[ 39%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq2_s.comp.cpp.o
[ 40%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_mxfp4.comp.cpp.o
[ 40%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q4_1.comp.cpp.o
[ 40%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_iq3_s.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q4_k.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q4_0.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q2_k.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q3_k.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q5_0.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q5_1.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q5_k.comp.cpp.o
[ 41%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q6_k.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/dequant_q8_0.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/diag_mask_inf.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/geglu.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/flash_attn_split_k_reduce.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/flash_attn_cm2.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/gelu.comp.cpp.o
[ 42%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/gelu_erf.comp.cpp.o
[ 43%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/geglu_quick.comp.cpp.o
[ 44%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/get_rows_quant.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/flash_attn_cm1.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/get_rows.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/div.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/geglu_erf.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/hardswish.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/im2col.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/group_norm.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/hardsigmoid.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/gelu_quick.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/flash_attn.comp.cpp.o
[ 45%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/l2_norm.comp.cpp.o
[ 46%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/im2col_3d.comp.cpp.o
[ 46%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/exp.comp.cpp.o
[ 46%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul.comp.cpp.o
[ 46%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/leaky_relu.comp.cpp.o
[ 46%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq1_m.comp.cpp.o
[ 47%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec.comp.cpp.o
[ 47%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_split_k_reduce.comp.cpp.o
[ 47%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq2_xs.comp.cpp.o
[ 47%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq2_s.comp.cpp.o
[ 48%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_q2_k.comp.cpp.o
[ 48%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq3_xxs.comp.cpp.o
[ 48%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_q4_k.comp.cpp.o
[ 48%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq1_s.comp.cpp.o
[ 48%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq3_s.comp.cpp.o
[ 49%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_iq2_xxs.comp.cpp.o
[ 49%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_nc.comp.cpp.o
[ 49%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_p021.comp.cpp.o
[ 49%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_q5_k.comp.cpp.o
[ 49%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_q3_k.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vecq.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mm_cm2.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mm.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mat_vec_q6_k.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/multi_add.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/opt_step_sgd.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/opt_step_adamw.comp.cpp.o
[ 50%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/mul_mmq.comp.cpp.o
[ 51%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/norm.comp.cpp.o
[ 51%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/pad.comp.cpp.o
[ 51%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/pool2d.comp.cpp.o
[ 52%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/quantize_q8_1.comp.cpp.o
[ 52%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/reglu.comp.cpp.o
[ 52%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/repeat.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/repeat_back.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/relu.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rms_norm_back.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rms_norm.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/roll.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rope_vision.comp.cpp.o
[ 53%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rms_norm_partials.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rope_neox.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/silu_back.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/soft_max_back.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/soft_max.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rope_norm.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/square.comp.cpp.o
[ 54%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/sigmoid.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/rope_multi.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/scale.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/silu.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/sin.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/swiglu.comp.cpp.o
[ 56%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/sum_rows.comp.cpp.o
[ 57%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/sqrt.comp.cpp.o
[ 55%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ssm_conv.comp.cpp.o
[ 57%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/tanh.comp.cpp.o
[ 57%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/sub.comp.cpp.o
[ 57%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/swiglu_oai.comp.cpp.o
[ 57%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ssm_scan.comp.cpp.o
[ 58%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/topk_moe.comp.cpp.o
[ 58%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/timestep_embedding.comp.cpp.o
[ 58%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/upscale.comp.cpp.o
[ 58%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/wkv6.comp.cpp.o
[ 58%] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/wkv7.comp.cpp.o
[ 58%] Linking CXX shared library ../../../bin/libggml-vulkan.so
[ 58%] Built target ggml-vulkan
[ 59%] Building CXX object ggml/src/CMakeFiles/ggml.dir/ggml-backend-reg.cpp.o
[ 59%] Linking CXX shared library ../../bin/libggml.so
[ 59%] Built target ggml
[ 59%] Building CXX object src/CMakeFiles/llama.dir/llama-adapter.cpp.o
[ 59%] Building CXX object src/CMakeFiles/llama.dir/llama-cparams.cpp.o
[ 59%] Building CXX object src/CMakeFiles/llama.dir/llama-batch.cpp.o
[ 59%] Building CXX object src/CMakeFiles/llama.dir/llama-graph.cpp.o
[ 60%] Building CXX object src/CMakeFiles/llama.dir/llama.cpp.o
[ 60%] Building CXX object src/CMakeFiles/llama.dir/llama-arch.cpp.o
[ 60%] Building CXX object examples/gguf/CMakeFiles/llama-gguf.dir/gguf.cpp.o
[ 60%] Building CXX object src/CMakeFiles/llama.dir/llama-grammar.cpp.o
[ 60%] Building CXX object src/CMakeFiles/llama.dir/llama-kv-cache.cpp.o
[ 61%] Building CXX object src/CMakeFiles/llama.dir/llama-memory-hybrid.cpp.o
[ 61%] Building CXX object src/CMakeFiles/llama.dir/llama-memory.cpp.o
[ 62%] Building CXX object examples/gguf-hash/CMakeFiles/llama-gguf-hash.dir/gguf-hash.cpp.o
[ 62%] Building CXX object src/CMakeFiles/llama.dir/llama-kv-cache-iswa.cpp.o
[ 63%] Building CXX object src/CMakeFiles/llama.dir/llama-context.cpp.o
[ 63%] Building CXX object src/CMakeFiles/llama.dir/llama-hparams.cpp.o
[ 64%] Building CXX object src/CMakeFiles/llama.dir/llama-impl.cpp.o
[ 63%] Building CXX object src/CMakeFiles/llama.dir/llama-chat.cpp.o
[ 64%] Building CXX object src/CMakeFiles/llama.dir/llama-io.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-model.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-memory-recurrent.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/unicode-data.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-sampling.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-model-saver.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-mmap.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-quant.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-model-loader.cpp.o
[ 65%] Building CXX object src/CMakeFiles/llama.dir/llama-vocab.cpp.o
[ 66%] Building CXX object src/CMakeFiles/llama.dir/unicode.cpp.o
[ 66%] Linking CXX executable ../../bin/llama-gguf-hash
[ 66%] Built target llama-gguf-hash
[ 66%] Linking CXX executable ../../bin/llama-gguf
[ 66%] Built target llama-gguf
[ 66%] Linking CXX shared library ../bin/libllama.so
[ 66%] Built target llama
[ 66%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd.cpp.o
[ 66%] Building C object tests/CMakeFiles/test-c.dir/test-c.c.o
[ 66%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd-audio.cpp.o
[ 66%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/clip.cpp.o
[ 66%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd-helper.cpp.o
[ 66%] Building CXX object common/CMakeFiles/common.dir/common.cpp.o
[ 66%] Building CXX object common/CMakeFiles/common.dir/arg.cpp.o
[ 66%] Building CXX object common/CMakeFiles/common.dir/chat-parser.cpp.o
[ 66%] Building CXX object common/CMakeFiles/common.dir/chat.cpp.o
[ 66%] Building CXX object common/CMakeFiles/common.dir/json-schema-to-grammar.cpp.o
[ 66%] Building CXX object examples/simple-chat/CMakeFiles/llama-simple-chat.dir/simple-chat.cpp.o
[ 66%] Building CXX object examples/simple/CMakeFiles/llama-simple.dir/simple.cpp.o
[ 67%] Building CXX object common/CMakeFiles/common.dir/console.cpp.o
[ 67%] Building CXX object common/CMakeFiles/common.dir/regex-partial.cpp.o
[ 67%] Building CXX object common/CMakeFiles/common.dir/llguidance.cpp.o
[ 67%] Building CXX object common/CMakeFiles/common.dir/json-partial.cpp.o
[ 67%] Building CXX object common/CMakeFiles/common.dir/log.cpp.o
[ 68%] Building CXX object common/CMakeFiles/common.dir/ngram-cache.cpp.o
[ 68%] Building CXX object common/CMakeFiles/common.dir/speculative.cpp.o
[ 68%] Building CXX object common/CMakeFiles/common.dir/sampling.cpp.o
[ 68%] Linking C executable ../bin/test-c
[ 68%] Built target test-c
[ 68%] Linking CXX executable ../../bin/llama-simple
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_data'
collect2: error: ld returned 1 exit status
make[2]: *** [examples/simple/CMakeFiles/llama-simple.dir/build.make:102: bin/llama-simple] Error 1
make[1]: *** [CMakeFiles/Makefile2:3428: examples/simple/CMakeFiles/llama-simple.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 69%] Linking CXX executable ../../bin/llama-simple-chat
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_rte_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_neox_f16_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_len'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_rte_data'
/usr/bin/ld: ../../bin/libggml-vulkan.so: undefined reference to `rope_norm_f16_data'
collect2: error: ld returned 1 exit status
make[2]: *** [examples/simple-chat/CMakeFiles/llama-simple-chat.dir/build.make:102: bin/llama-simple-chat] Error 1
make[1]: *** [CMakeFiles/Makefile2:3458: examples/simple-chat/CMakeFiles/llama-simple-chat.dir/all] Error 2
[ 70%] Linking CXX shared library ../../bin/libmtmd.so
[ 70%] Built target mtmd
[ 70%] Linking CXX static library libcommon.a
[ 70%] Built target common
make: *** [Makefile:146: all] Error 2
```

--- a/ggml/src/ggml-vulkan/vulkan-shaders/vulkan-shaders-gen.cpp
+++ b/ggml/src/ggml-vulkan/vulkan-shaders/vulkan-shaders-gen.cpp
@@ -317,7 +317,8 @@ void string_to_spv_func(std::string name, std::string in_path, std::string out_p
 
     // disable spirv-opt for coopmat shaders for https://github.com/ggerganov/llama.cpp/issues/10734
     // disable spirv-opt for bf16 shaders for https://github.com/ggml-org/llama.cpp/issues/15344
-    std::string opt_level = (coopmat || name.find("bf16") != std::string::npos) ? "" : "-O";
+    // disable spirv-opt for rope shaders for https://github.com/ggml-org/llama.cpp/issues/16860
+    std::string opt_level = (coopmat || name.find("bf16") != std::string::npos || name.find("rope") != std::string::npos) ? "" : "-O";
 
     #ifdef _WIN32
         std::vector<std::string> cmd = {GLSLC, "-fshader-stage=compute", target_env, opt_level, "\"" + in_path + "\"", "-o", "\"" + out_path + "\""};

